src/vs/workbench/services/positronConsole/browser/positronConsoleService.ts

```diff
@@ -2081,36 +2084,48 @@ class PositronConsoleInstance extends Disposable implements IPositronConsoleInst
 	/**
 	 * Executes code.
 	 * @param code The code to execute.
+	 * @param runtimeCodeExecutionMode Possible code execution modes for a language runtime
+	 * runtimeCodeExecutionMode
 	 */
-	private doExecuteCode(code: string) {
+	private doExecuteCode(code: string, runtimeCodeExecutionMode?: RuntimeCodeExecutionMode) {
 		// Create the ID for the code that will be executed.
 		const id = `fragment-${generateUuid()}`;
 
-		// Create the provisional ActivityItemInput.
-		const activityItemInput = new ActivityItemInput(
-			ActivityItemInputState.Provisional,
-			id,
-			id,
-			new Date(),
-			this._session.dynState.inputPrompt,
-			this._session.dynState.continuationPrompt,
-			code
-		);
+		// Fallback to an interactive exection mode if one is not provided
+		const codeExecutionMode = runtimeCodeExecutionMode || RuntimeCodeExecutionMode.Interactive;
 
-		// Add the provisional ActivityItemInput. This provisional ActivityItemInput will be
-		// replaced with the real ActivityItemInput when the runtime sends it (which can take a
-		// moment or two to happen).
-		this.addOrUpdateUpdateRuntimeItemActivity(id, activityItemInput);
+		// If the code exection mode is silent, skip creating a runtimeItem
+		// to prevent the code input from being exposed in the console UI
+		if (codeExecutionMode !== RuntimeCodeExecutionMode.Silent) {
+			// Create the provisional ActivityItemInput.
+			const activityItemInput = new ActivityItemInput(
+				ActivityItemInputState.Provisional,
+				id,
+				id,
+				new Date(),
+				this._session.dynState.inputPrompt,
+				this._session.dynState.continuationPrompt,
+				code
+			);
+
+			// Add the provisional ActivityItemInput. This provisional ActivityItemInput will be
+			// replaced with the real ActivityItemInput when the runtime sends it (which can take a
+			// moment or two to happen).
+			this.addOrUpdateUpdateRuntimeItemActivity(id, activityItemInput);
+		}
 
 		// Execute the code.
 		this.session.execute(
 			code,
 			id,
-			RuntimeCodeExecutionMode.Interactive,
+			codeExecutionMode,
 			RuntimeErrorBehavior.Continue);
 
-		// Fire the onDidExecuteCode event.
-		this._onDidExecuteCodeEmitter.fire(code);
+		// Code that is not executed interactively should not show up in the console history
+		if (codeExecutionMode === RuntimeCodeExecutionMode.Interactive) {
+			// Fire the onDidExecuteCode event.
+			this._onDidExecuteCodeEmitter.fire(code);
+		}
```

Comment:
I think this is worth at least taking a quick look at, to see if it is doable.

Reply:
I had a similar thought and was wondering if that is the route to go but didn't initially add it because it seems like this service is mainly used to drive information the user sees (I could totally be wrong about this though). I agree with this feeling like a footgun so I'll update this so it includes the mode!